lib/client.js

```diff
@@ -190,9 +190,9 @@ function MqttClient (streamBuilder, options) {
   })
 
   // resubscribe
-  this.on('connect', function () {
-    if (this.options.clean && Object.keys(this._subscribedTopics).length > 0) {
-      this.subscribe(this._subscribedTopics)
+  this.on('reconnect', function () {
```

Comment:
That case is covered in the test I added already: https://github.com/mqttjs/MQTT.js/pull/556/files#diff-7e3bd8e7059a21bfd255649cbfc84b33R411

Do you want it in a separate test as well?